File: Udemy-Neal-Davis/section-4-elb/hol-59-scaling-policy/generate-load.js
const dnsName = "AppLoadBalancer-977078372.eu-west-2.elb.amazonaws.com";
const websiteUrl = `http://${dnsName}`;

const maxRequest = 100000;
let requestCount = 0;
const repeat = setInterval(() => {
  requestCount++;
  fetch(websiteUrl)
    .then((res) => res.text())
    .then((html) => console.log(`Result ${requestCount}: ${html}`))
    .catch((error) => console.log("Err", error));

  if (requestCount > maxRequest) {
    clearInterval(repeat);
    process.exit(0);
  }
}, 200);

File: Udemy-Neal-Davis/section-4-elb/hol-59-scaling-policy/generate-load.py
import threading
import requests
dns_name = 'AppLoadBalancer-977078372.eu-west-2.elb.amazonaws.com' 
website_url = 'http://' + dns_name

max_request = 100000
request_count = 0

def make_request():
  global request_count
  request_count = request_count + 1
  print(f"Making Request {request_count}")
  if (request_count > max_request):
    return

  response = requests.get(website_url)
  if response.status_code == 200 : 
    print(f"Html: {response.text}")

  threading.Timer(2, make_request).start()

make_request()
